### PR TITLE
HDDS-1148. After allocating container, we are not adding to container DB.

### PR DESCRIPTION
**Pull request by bharatviswa504**

### CODE REVIEW

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/SCMContainerManager.java [hunk 1]
```diff
@@ -360,16 +355,108 @@ public ContainerInfo getMatchingContainer(final long sizeRequired,
...
+          // Add to DB
+          addContainerToDB(containerInfo);
+          containerStateManager.updateLastUsedMap(pipeline.getId(),
+              containerInfo.containerID(), owner);
+        }
+      } else {
+        containerStateManager.updateLastUsedMap(pipeline.getId(),
+            containerInfo.containerID(), owner);
+      }
+      // TODO: #CLUTIL cleanup entries in lastUsedMap
```
Comment (nandakumar131) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/SCMContainerManager.java]: containerStateManager.updateLastUsedMap can be called just before the return statement and avoided in multiple places.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/SCMContainerManager.java [hunk 2]
```diff
@@ -215,23 +222,11 @@ public ContainerInfo allocateContainer(final ReplicationType type,
       throws IOException {
     lock.lock();
     try {
-      final ContainerInfo containerInfo; containerInfo = containerStateManager
+      final ContainerInfo containerInfo;
```
Comment (nandakumar131) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/SCMContainerManager.java]: Can be put as single statement --> 
final ContainerInfo containerInfo = containerStateManager.allocateContainer(pipelineManager, type, replicationFactor, owner);

**Comment by apache-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 26 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1340 | trunk passed |
| -1 | compile | 28 | server-scm in trunk failed. |
| -0 | checkstyle | 17 | The patch fails to run checkstyle in server-scm |
| -1 | mvnsite | 13 | server-scm in trunk failed. |
| +1 | shadedclient | 734 | branch has no errors when building and testing our client artifacts. |
| -1 | findbugs | 13 | server-scm in trunk failed. |
| -1 | javadoc | 21 | server-scm in trunk failed. |
||| _ Patch Compile Tests _ |
| -1 | mvninstall | 11 | server-scm in the patch failed. |
| -1 | compile | 10 | server-scm in the patch failed. |
| -1 | javac | 10 | server-scm in the patch failed. |
| -0 | checkstyle | 9 | The patch fails to run checkstyle in server-scm |
| -1 | mvnsite | 11 | server-scm in the patch failed. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 791 | patch has no errors when building and testing our client artifacts. |
| -1 | findbugs | 12 | server-scm in the patch failed. |
| -1 | javadoc | 13 | server-scm in the patch failed. |
||| _ Other Tests _ |
| -1 | unit | 15 | server-scm in the patch failed. |
| +1 | asflicense | 27 | The patch does not generate ASF License warnings. |
| | | 3191 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-511/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/511 |
| Optional Tests |  dupname  asflicense  compile  javac  javadoc  mvninstall  mvnsite  unit  shadedclient  findbugs  checkstyle  |
| uname | Linux 4426ca5c8520 4.4.0-139-generic #165~14.04.1-Ubuntu SMP Wed Oct 31 10:55:11 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / b17a260 |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/1/artifact/out/branch-compile-hadoop-hdds_server-scm.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/1/artifact/out//testptch/patchprocess/maven-branch-checkstyle-hadoop-hdds_server-scm.txt |
| mvnsite | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/1/artifact/out/branch-mvnsite-hadoop-hdds_server-scm.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/1/artifact/out/branch-findbugs-hadoop-hdds_server-scm.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/1/artifact/out/branch-javadoc-hadoop-hdds_server-scm.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/1/artifact/out/patch-mvninstall-hadoop-hdds_server-scm.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/1/artifact/out/patch-compile-hadoop-hdds_server-scm.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/1/artifact/out/patch-compile-hadoop-hdds_server-scm.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/1/artifact/out//testptch/patchprocess/maven-patch-checkstyle-hadoop-hdds_server-scm.txt |
| mvnsite | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/1/artifact/out/patch-mvnsite-hadoop-hdds_server-scm.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/1/artifact/out/patch-findbugs-hadoop-hdds_server-scm.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/1/artifact/out/patch-javadoc-hadoop-hdds_server-scm.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/1/artifact/out/patch-unit-hadoop-hdds_server-scm.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/1/testReport/ |
| Max. process+thread count | 339 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/server-scm U: hadoop-hdds/server-scm |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/SCMContainerManager.java (hunk 1 above)
Comment (bharatviswa504) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/SCMContainerManager.java]: Done

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/SCMContainerManager.java (hunk 2 above)
Comment (bharatviswa504) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/SCMContainerManager.java]: Done

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/SCMContainerManager.java [hunk 3]
```diff
@@ -360,16 +355,108 @@ public ContainerInfo getMatchingContainer(final long sizeRequired,
       String owner, Pipeline pipeline) {
     try {
       //TODO: #CLUTIL See if lock is required here
-      return containerStateManager
-          .getMatchingContainer(sizeRequired, owner, pipelineManager,
-              pipeline);
+      NavigableSet<ContainerID> containerIDs =
+          pipelineManager.getContainersInPipeline(pipeline.getId());
+      if (containerIDs == null) {
```
[the remark this answers is not in the document]
Comment (bharatviswa504) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/SCMContainerManager.java]: Done

**Comment by apache-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 41 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1068 | trunk passed |
| -1 | compile | 25 | server-scm in trunk failed. |
| -0 | checkstyle | 14 | The patch fails to run checkstyle in server-scm |
| -1 | mvnsite | 15 | server-scm in trunk failed. |
| +1 | shadedclient | 722 | branch has no errors when building and testing our client artifacts. |
| -1 | findbugs | 15 | server-scm in trunk failed. |
| -1 | javadoc | 17 | server-scm in trunk failed. |
||| _ Patch Compile Tests _ |
| -1 | mvninstall | 11 | server-scm in the patch failed. |
| -1 | compile | 10 | server-scm in the patch failed. |
| -1 | javac | 10 | server-scm in the patch failed. |
| -0 | checkstyle | 9 | The patch fails to run checkstyle in server-scm |
| -1 | mvnsite | 11 | server-scm in the patch failed. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 764 | patch has no errors when building and testing our client artifacts. |
| -1 | findbugs | 14 | server-scm in the patch failed. |
| -1 | javadoc | 13 | server-scm in the patch failed. |
||| _ Other Tests _ |
| -1 | unit | 15 | server-scm in the patch failed. |
| +1 | asflicense | 27 | The patch does not generate ASF License warnings. |
| | | 2889 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-511/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/511 |
| Optional Tests |  dupname  asflicense  compile  javac  javadoc  mvninstall  mvnsite  unit  shadedclient  findbugs  checkstyle  |
| uname | Linux 29030c11742b 3.13.0-153-generic #203-Ubuntu SMP Thu Jun 14 08:52:28 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / d33f066 |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/2/artifact/out/branch-compile-hadoop-hdds_server-scm.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/2/artifact/out//testptch/patchprocess/maven-branch-checkstyle-hadoop-hdds_server-scm.txt |
| mvnsite | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/2/artifact/out/branch-mvnsite-hadoop-hdds_server-scm.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/2/artifact/out/branch-findbugs-hadoop-hdds_server-scm.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/2/artifact/out/branch-javadoc-hadoop-hdds_server-scm.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/2/artifact/out/patch-mvninstall-hadoop-hdds_server-scm.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/2/artifact/out/patch-compile-hadoop-hdds_server-scm.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/2/artifact/out/patch-compile-hadoop-hdds_server-scm.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/2/artifact/out//testptch/patchprocess/maven-patch-checkstyle-hadoop-hdds_server-scm.txt |
| mvnsite | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/2/artifact/out/patch-mvnsite-hadoop-hdds_server-scm.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/2/artifact/out/patch-findbugs-hadoop-hdds_server-scm.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/2/artifact/out/patch-javadoc-hadoop-hdds_server-scm.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/2/artifact/out/patch-unit-hadoop-hdds_server-scm.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/2/testReport/ |
| Max. process+thread count | 339 (vs. ulimit of 5500) |
| modules | C: hadoop-hdds/server-scm U: hadoop-hdds/server-scm |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-511/2/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request merged**